azure-pipelines.yaml

```diff
@@ -0,0 +1,201 @@
+variables:
+    azureContainerRegistry: jwminesupport
+    azureContainerRegistryFull: jwminesupport.azurecr.io
+    azureSubscriptionEndpoint: MineSupport
+    azureResourceGroup: MineSupport
+    kubernetesCluster: MineSupport
+    imageName:  mine-support-payment-service
+    containerTag: $(Build.SourceBranchName)
+    safeContainerTag: $(Build.SourceBranchName)
+    mergedPrNo:
+steps:
+  # extracts PR number from refs/pull/{prNo}/ branch name and sets to containerTag if pull request
+- script: |
+    IFS='/' read -r -a array <<< $BUILD_SOURCEBRANCH
+    echo '##vso[task.setvariable variable=containerTag]pr'${array[2]}
+  condition: and(succeeded(), contains(variables['Build.SourceBranch'], 'refs/pull'))
+  displayName: set container tag if PR
+  # extracts version number from refs/tags/{versionNo}/ branch name and sets to containerTag if release
+- script: |
+    IFS='/' read -r -a array <<< $BUILD_SOURCEBRANCH
+    echo '##vso[task.setvariable variable=containerTag]'${array[2]}
+  condition: and(succeeded(), contains(variables['Build.SourceBranch'], 'refs/tags'))
+  displayName: set container tag if release
+  # update helm charts tag and version if files ready for publishing chart
+- script: |
+    sed -i -e 's/image: mine-support/image: mine-support:$(containerTag)/' ./helm/values.yaml
+    versionTag=`echo $(containerTag) | sed 's/v//'`
+    sed -i -e 's/version: 1.0.0/version: '"$versionTag"'/' ./helm/Chart.yaml
+  condition: and(succeeded(), contains(variables['Build.SourceBranch'], 'refs/tags'))
+  displayName: update values and Chart yaml for release
+  # create safe container tag for use in namespaces
+- script: |
+    safeContainerTag=`echo $(containerTag) | sed 's/[^a-zA-Z0-9]/-/g' | tr '[:upper:]' '[:lower:]'`
+    echo '##vso[task.setvariable variable=safeContainerTag]'$safeContainerTag
+  displayName: set safe container tag as lower case container tag with illegal characters removed
+  # extracts merged PR number from git log if a PR merge to master
+- script: |
+    gitLog=`git log --pretty=oneline --abbrev-commit -1`
+    mergedPrNo=`echo $gitLog | sed -n 's/.*(#\([0-9]\+\)).*/\1/p'`
+    echo '##vso[task.setvariable variable=mergedPrNo]'$mergedPrNo
+  condition: and(succeeded(), contains(variables['Build.SourceBranch'], format('refs/heads/{0}', variables['Build.SourceBranchName'])))
+  displayName: extracts merged PR number if merge to branch
+  # create test folder
+- script: |
+    mkdir test-output
+    chmod 777 test-output
+  displayName: create test output directory
+  # build test image
+- task: DockerCompose@0
+  inputs:
+    containerregistrytype: Azure Container Registry
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureContainerRegistry: $(azureContainerRegistryFull)
+    dockerComposeFile: docker-compose.yaml
+    projectName: $(Build.Repository.Name)
+    qualifyImageNames: true
+    dockerComposeCommand: -f docker-compose.test.yaml build --no-cache $(imageName)

```

Comment:
It would be a little cleaner to use the `Build services` action and the `additionalDockerComposeFiles` option for the Azure Docker Compose task. Other services have been modified to use the following in the `full-stack-development` branches:
```
  # build test image
- task: DockerCompose@0
  inputs:
    action: Build services
    containerregistrytype: Azure Container Registry
    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
    azureContainerRegistry: $(azureContainerRegistryFull)
    dockerComposeFile: docker-compose.yaml
    additionalDockerComposeFiles: docker-compose.test.yaml
    projectName: $(Build.Repository.Name)
    qualifyImageNames: true
  displayName: build test image
```

Reply:
Updated as request